client/sections-middleware.js

```diff
@@ -40,7 +35,8 @@ async function loadSection( context, sectionDefinition ) {
 		// call the module initialization function (possibly async, registers page.js handlers etc.)
 		await requiredModule.default( controller.clientRouter, addReducerToStore( context.store ) );
 	} finally {
-		context.store.dispatch( { type: 'SECTION_SET', isLoading: false } );
+		// Do not update the current active section, only change `isLoading` flag
+		context.store.dispatch( setSection( undefined, { isLoading: true } ) );
```

Comment:
Did you mean to make this `isLoading: false`? By setting it to `true` here I believe it'll always be stuck in a loading state.

Reply:
Well spotted, totally a copy/paste mistake